flycheck.el

```diff
@@ -7684,7 +7684,25 @@ See URL `https://github.com/jimhester/lintr'."
           line-end))
   :modes ess-mode
   ;; Don't check ESS files which do not contain R
-  :predicate (lambda () (equal ess-language "S")))
+  :predicate (lambda () (and (equal ess-language "S")
+                        (flycheck-R-has-lintr (flycheck-checker-executable 'r-lintr))))
+  :verify (lambda (checker)
+            (let ((has-lintr (flycheck-R-has-lintr
+                              (flycheck-checker-executable checker))))
+              (list
+               (flycheck-verification-result-new
+                :label "lintr library"
+                :message (if has-lintr "present" "missing")
+                :face (if has-lintr 'success '(bold error)))))))
+
+(defun flycheck-R-has-lintr (R)
+  "Whether R has installed the `lintr' library."
+  (with-temp-buffer
+    (call-process R nil t nil
+                  "--slave" "--restore" "--no-save" "-e"
+                  "library('lintr')")
+    (goto-char (point-min))
+    (not (looking-at-p "Error in library(\"lintr\") : there is no package called ‘lintr’"))))
```

Comment:
yes they are smart quotes, but can be disabled adding to `LC_ALL=C` the process-environment

```elisp

(let ((process-environment (append '("LC_ALL=C") process-environment)))
  ;; ...
  ;; or simply check for "there is no package called"  
  (not (re-search-forward "there is no package called 'lintr'" nil 'no-error)) 
```